pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,14 @@
+package agent
+
+// Platform defines agent based install configuration specific to bare metal clusters.
+type Platform struct {
+	// APIVIP is the virtual IP used to reach the OpenShift cluster's API.
+	APIVIP string `json:"apiVIP"`
+	// APIVIPDNSName is the domain name used to reach the OpenShift cluster API.
+	APIVIPDNSName string `json:"apiVIPDNSName"`
+	// IngressVIP is the virtual IP used for cluster ingress traffic.
+	IngressVIP string `json:"ingressVIP"`
+	// VIPDHCPAllocation indicates if virtual IP DHCP allocation mode is enabled.
+	// +optional
+	VIPDHCPAllocation bool `json:"vipDHCPAllocation"`
```

Comment:
personally i would recommend not using `boolean`, this types of field almost always end up growing in valid states.

make it a string enum,
```
Disabled ""
Enabled "Enabled"
```
to capture the same default of empty type value and expand later on.

Reply:
Good idea.